modules/pam_mkhomedir/mkhomedir_helper.c

```diff
@@ -39,6 +41,7 @@ create_homedir(const struct passwd *pwd,
    DIR *d;
    struct dirent *dent;
    int retval = PAM_SESSION_ERR;
+   mode_t mode = umask(0777);
```

Comment:
This `umask` invocation actually _changes_ the process umask.
I don't follow why do you need to obtain the process umask,
and why do you consider appropriate changing it.

Reply:
With new modifications, it doesn't deal with `umask` anymore